addons/website_event/controllers/main.py

```diff
@@ -312,7 +312,27 @@ def registration_new(self, event, **post):
                 availability_check = False
         if not tickets:
             return False
-        return request.env['ir.ui.view']._render_template("website_event.registration_attendee_details", {'tickets': tickets, 'event': event, 'availability_check': availability_check})
+        default_first_attendee = {}
+        visitor = request.env['website.visitor']._get_visitor_from_request()
```

Comment:
To save some queries, I would fertch it only if env.user is public. That way we don't fetch visitor again if we don't need it.